### DISCUSSION TITLE
How do I get VSCodium to use my $PATH variables without launching it from the terminal?

### DISCUSSION THREAD

**Discussion by sylvesterroos** [VSCodium/vscodium#1243] [Q&A] [answered]
Doom Emacs solves this problem by creating an `env` after reading all the variables in my shell, and reads this `env` file when it starts so that it behaves like when launching it from the terminal. Does VSCodium have this sort of functionality? If not, would it be feasible to implement it?

(I'm not sure about the last part considering that VSCodium is little more than FOSS VSCode. Would it be better to ask this question in the VSCode repo?)

**Comment by daiyam** [comment, marked as answer]
@Pistrie I'm assuming that you are on Linux, no? What is your distro? Which package manager have you used to install codium?

**Reply by sylvesterroos** [reply]
Yes I'm on Linux, and my distro is EndeavourOS. I installed the [vscodium](https://aur.archlinux.org/packages/vscodium) package from the AUR.

**Reply by daiyam** [reply]
I don't know the details of how the env variables are populated in the integrated terminal of Visual Studio Code.
But there shouldn't have any problems from the AUR package. Do you have any issue?
The entry script is https://github.com/microsoft/vscode/blob/main/resources/linux/bin/code.sh, not sure if it can help for what you are looking.

**Reply by sylvesterroos** [reply]
Nono the problem is not with the integrated terminal.

I'm programming in Elixir, and the Elixir binary is not in the default location (`/usr/bin`) because I installed it with the asdf version manager. The Elixir language server doesn't work work because it can't find the Elixir binary in the default path. In my shell config I add this alternative path to the $PATH variable, which is why GUI applications not launched from the shell have the incomplete $PATH

**Reply by daiyam** [reply]
Maybe this can help you: https://github.com/elixir-lsp/elixir-ls/issues/665
I do find `asdf` to be, some time, a bit troublesome...

**Reply by sylvesterroos** [reply]
I'm going off-topic now, but what do you use, if not the latest version of your package manager? asdf has worked without any real problems for me
